packages/plugin-azuremaps/src/main/java/com/openmobilehub/android/maps/plugin/azuremaps/presentation/maps/OmhMarkerImpl.kt

```diff
@@ -321,7 +326,7 @@ internal class OmhMarkerImpl(
         val markerImageID = getMarkerIconID(icon != null)
 
         // ensure the other custom icon is removed for memory optimization (if present)
-        mapViewDelegate.removeImage(getMarkerIconID(!isCustomIconSet))
+        if (lastMarkerIconID != null) mapViewDelegate.removeImage(lastMarkerIconID!!)
```

Comment:
```suggestion
        lastMarkerIconID?.let { mapViewDelegate.removeImage(it) }
```

Reply:
Refactored in 3203a66940400da0906cdf14482f0856d617ee7a